src/mono/mono/metadata/loader-internals.h

```diff
@@ -62,6 +72,53 @@ struct _MonoAssemblyLoadContext {
 };
 #endif /* ENABLE_NETCORE */
 
+struct _MonoMemoryManager {
+	// Whether the MemoryManager can be unloaded on netcore; should only be set at creation
+	gboolean collectible;
+	// Whether this is a singleton or generic MemoryManager
+	gboolean is_generic;
+	// Whether the MemoryManager is in the process of being freed
+	gboolean freeing;
+
+	// Entries moved over from the domain:
+
+	// If taking this with the loader lock, always take this second
+	// On legacy, this does _not_ protect mp/code_mp, which are covered by the domain lock
+	MonoCoopMutex lock;
+
+	MonoMemPool *mp;
+	MonoCodeManager *code_mp;
+
+	GPtrArray *class_vtable_array;
+
+	// !!! REGISTERED AS GC ROOTS !!!
+	// Hashtables for Reflection handles
```

Comment:
Where are these registered as roots ?

Reply:
`mono_g_hash_table_new_type_internal` registers them at initialization, see memory-manager.c

Reply:
The memory around pointing to those needs to be registered with the GC, see the domain_gc_desc code in mono_domain_create ().

Reply:
Is that actually needed? I know that's used to make them gc-allocated, which I thought we didn't need here, but shouldn't they be registered as roots regardless?

Reply:
We pass `MONO_HASH_VALUE_GC` so it should call `mono_gc_register_root_wbarrier`, what else do I need to do? cc: @BrzVlad

Reply:
I'm pretty sure this was needed before, but maybe not anymore, since MonoGHashTable is now allocated using g_malloc ().